core/service/storage-service.js

```diff
@@ -108,46 +108,51 @@ var StorageService = exports.StorageService = Montage.specialize({
-                this._dataService.getNewInstanceForType(Model.VolumeDatasetProperties).then(function(newProperties) {
+                return this._dataService.getNewInstanceForType(Model.VolumeDatasetProperties).then(function(newProperties) {
+                    var promises = [];
                     dataset.properties = newProperties;
-
-                    self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyAtime).then(function(newAtime) {
-                        newAtime.source = "INHERITED";
-                        dataset.properties.atime = newAtime;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyCasesensitivity);
-                    }).then(function(newCasesensitivity) {
-                        newCasesensitivity.source = "INHERITED";
-                        dataset.properties.casesensitivity = newCasesensitivity;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyCompression);
-                    }).then(function(newCompression) {
-                        newCompression.source = "INHERITED";
-                        dataset.properties.compression = newCompression;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyDedup);
-                    }).then(function(newDedup) {
-                        newDedup.source = "INHERITED";
-                        dataset.properties.dedup = newDedup;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyQuota);
-                    }).then(function(newQuota) {
-                        dataset.properties.quota = newQuota;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyRefquota);
-                    }).then(function(newRefquota) {
-                        dataset.properties.refquota = newRefquota;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyVolblocksize);
-                    }).then(function(newVolblocksize) {
-                        newVolblocksize.parsed = 512;
-                        dataset.properties.volblocksize = newVolblocksize;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyRefreservation);
-                    }).then(function(newRefreservation) {
-                        dataset.properties.refreservation = newRefreservation;
-                        return self._dataService.getNewInstanceForType(Model.VolumeDatasetPropertyReservation);
-                    }).then(function(newReservation) {
-                        dataset.properties.reservation = newReservation;
-                    });
+                    promises.push(
+                        self._populateDatasetPropertyModel(dataset, "VolumeDatasetPropertyAtime", "atime").then(function() {
+                            dataset.properties.atime.source = "INHERITED";
```

Comment:
Isn't this the same as https://github.com/freenas/gui/blob/master/core/model/enumerations/volume-property-source.js#L3 ?
If it is, you should use the enumeration value here. If it is not, you should use a constant for all constants strings.